contracts/interfaces/IAelinUpFrontDeal.sol

```diff
@@ -0,0 +1,64 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.6;
+
+import {IAelinPool} from "./IAelinPool.sol";
+
+interface IAelinUpFrontDeal {
+    struct UpFrontPool {
+        string name;
+        string symbol;
+        uint256 purchaseTokenCap;
+        address purchaseToken;
+        uint256 sponsorFee;
+        uint256 purchaseDuration;
+        address[] allowListAddresses;
+        uint256[] allowListAmounts;
+        IAelinPool.NftCollectionRules[] nftCollectionRules;
+    }
+
+    struct UpFrontDeal {
+        address underlyingDealToken;
+        uint256 underlyingDealTokenTotal;
+        uint256 vestingPeriod;
+        uint256 vestingCliffPeriod;
+        uint256 proRataRedemptionPeriod;
+        address holder;
+        uint256 maxDealTotalSupply;
+    }
```

Comment:
Thinking we can make these two structs one.

Add a sponsor address, purchaseTokensPerDealTokens, 

Thinking we can remove purchaseTokenCap (will be capped by the maxDealTotalSupply and the purchaseTokensPerDealTokens), and proRataRedemptionPeriod (purchase automatically accepts)

Is there a reason for both the maxDealTotalSupply and the underlyingDealTokenTotal?

Reply:
no, there should not be 2 values here